src/Contexts/PostsContext.jsx

```diff
@@ -19,57 +19,21 @@ export const PostsProvider = ({ children }) => {
 
   useEffect(() => {
     const getPosts = () => {
-      api
-        .get("/posts?_start=0&_limit=20")
-        .then((res) => {
-          setPosts(res.data);
-        })
-        .catch((err) => {
-          console.log("Failed to load 0~20 posts", err);
-        })
-        .then(
-          api
-            .get("/posts?_start=20&_limit=20")
-            .then((res) => {
+      api.get("/posts?_start=0&_limit=20").then((res) => {
+        setPosts(res.data);
+        api.get("/posts?_start=20&_limit=20").then((res) => {
+            setPosts((posts) => [...posts, ...res.data]);
```

Comment:
Aqui eu trocaria de `posts` para `oldPosts` ou `oldState`, talvez até `currentPosts`. Eu vejo utilizaram bastante `oldSomething`

```suggestion
            setPosts((oldPosts) => [...oldPosts, ...res.data]);
```